### DISCUSSION TITLE
Issue with Date Serialization/Deserialization in Tests using useFetch

### DISCUSSION THREAD

**Discussion by bchanez** [nuxt/nuxt#27377] [Questions] [answered]
Hello everyone,

I'm encountering an issue with the serialization and deserialization of Date objects when using useFetch in my tests. I have a mocked endpoint that returns a date, but the date is serialized as a string. When I retrieve the response, it comes back as a string instead of a Date instance.

Here is the code I'm using:

```ts
  async function testDate() {
    const { data } = await useFetch('api/test')
    return data.value
  }

 it('test', async () => {
    // given
    const input = new Date()
    registerEndpoint('api/test', () => (
      input
    ))

    // when
    const response = await testDate()

    console.log(response)
    console.log(typeof response)
    console.log(response instanceof Date)

    // then
    expect(response).toStrictEqual(input)
  })
```
  Here is the output of my test:
```
expected '2024-05-27T19:05:24.671Z' to strictly equal 2024-05-27T19:05:24.671Z

2024-05-27T19:05:24.671Z
string
false
```
As shown, response is still a string, not a Date object.

Thanks in advance for your help!

**Comment by TheAlexLichter** [comment, marked as answer]
That's correct, as API responses are realized and return the "string version" of an object (commonly JSON for plain objects and classes). Dates are converted to their string representation :) 

It is your task to either change it back via `transform` or use e.g. a [payload reviver (more common for classes)](https://www.youtube.com/watch?v=8w6ffRBs8a4)

**Reply by SoaresMG** [reply]
Is there a way to switch the whole reduce/revive with superjson?

I tried it myself and made it work but it's a real pain to make it generic enough for all useFetch calls while removing SerializeObject.

**Reply by SoaresMG** [reply]
I have revisited this again and I can't find much info about it.
It would be awesome to have `superjson` support so you can have complex types being automatically revived/reduced
